src/generators/markdowndocgenerator/toolbargenerator.tsx

```diff
@@ -0,0 +1,64 @@
+import { MarkdownDocGeneratorOptionsManager } from './optionsmanager';
+
+import * as React from 'react';
+
+interface NotebookGeneratorToolbarProps {}
+
+interface NotebookGeneratorToolbarState {
+  numbering: boolean;
+}
+
+export function markdownDocGeneratorToolbar(
+  options: MarkdownDocGeneratorOptionsManager
+) {
+  // Render the toolbar
+  return class extends React.Component<
+    NotebookGeneratorToolbarProps,
+    NotebookGeneratorToolbarState
+  > {
+    constructor(props: NotebookGeneratorToolbarProps) {
+      super(props);
+      this.state = { numbering: true };
+      options.initializeOptions(true);
+    }
+
+    toggleAutoNumbering = () => {
+      options.numbering = !options.numbering;
+      this.setState({ numbering: options.numbering });
+    };
+
+    render() {
+      let numberingIcon = this.state.numbering ? (
+        <div
+          className="toc-toolbar-auto-numbering-button toc-toolbar-button"
+          onClick={event => this.toggleAutoNumbering()}
+        >
+          <img
+            alt="Toggle Auto-Numbering"
+            title="Toggle Auto-Numbering"
+            src={require('../../../style/img/autonumbering_selected.svg')}
```

Comment:
The convention we have been using elsewhere is to set icon images using CSS and a `background-image` property. This makes the webpackification a bit simpler.